app/assets/javascripts/common/password_validation.js

```diff
@@ -39,15 +30,41 @@ PasswordValidation = function() {
         "X-CSRF-Token": token
       },
       data: {"password": password}
-    }).done(function(data, status){
-      console.log(status)
-      console.log(data)
-      this.timer = null;
+    }).done((data, status) => {
+      let response;
       if (status === "success") {
-        this.response = data["errors"];
+        response = data["errors"];
       } else {
-        this.response = null;
+        // if we don't get a response, mark all validations as failures
+        response = Validations;
       }
+      this.updateChecklist(response);
+      this.updateSubmitStatus(response);
+    });
+  }
+
+  this.updateChecklist = (response) => {
+    Validations.forEach(validationName => {
+      response.includes(validationName) ? this.uncheckItem(validationName) : this.checkItem(validationName);
     });
```

Comment:
I think this may be more clear to just do add/remove classes inline - this is all easy enough to follow and at roughly the same level of abstraction as the `updateSubmitStatus` function:
```suggestion
    Validations.forEach(validation => {
      if(response.includes(validation)) {
        $(`#$(validation)`).removeClass("completed")
      } else {
        $(`#$(validation)`).addClass("completed")  
      }
   });
```

Reply:
Done.